### ISSUE TITLE
Free space check before downloading a file should take free space limit into account.

### ISSUE THREAD

**Issue by nekr0z** [syncthing/syncthing#4574]
Currently there's a per-file check that looks if file that is about to be downlowaded would fit onto the filesystem. This check doesn't take free space limit (currently 1% by default) into account while assessing available space. It should.
This would not fix #2370 but would in many cases help prevent the situation discussed there.

Sample situation: 20 MB free on a 1 TB filesystem (free space limit is 1% = 10 MB). Files A and B are 15 MB each, files C and D are 30 MB each.
Day 1: A and B are added globally. Day 2: C and D are deleted globally.

CURRENT BEHAVIOUR:
Day 1. A is downloaded, 5 MB of free space is left. B is not downloaded, because it's bigger than free space available. Then check for a free space limit is performed, 5 MB is less than 10 MB limit. The folder enters _Stopped_ state.
Day 2. Nothing happens, folder still _Stopped_. User intervention on local filesystem is required.

DESIRED BEHAVIOUR:
Day 1. Downloading A would result in 5 MB of free space which is less than the limit of 10 MB, so A is not downloaded, and neither is B. The folder enters _Out of sync_ state.
Day 2. C and D get deleted locally, now there's 80 MB free space, so A and B get downloaded leaving 50 MB of free space. Folder is now _Up to date_, operations continue normally, no user intervention required.

**Comment by calmh** [comment]
The per file check is to ensure we have space for the temp file. We know we need 5 MB of temp space to download a 5 MB file. We don't know the effect on free space after completing the operation though - we might lose 5 MB if the file was new or there is versioning, we might require no extra space if we replaced a 5 MB existing file without versioning, we might regain arbitrary amounts of space if we replaced a larger file. Potentially we could try to figure out the end result, but I can guarantee we would get it wrong in a bunch of corner cases so I don't think it's worth it.

You might argue that we should never stop the folder though, just stay out of sync and periodically retry - that's the behaviour introduced in your case described above with the suggested change.
